### ISSUE TITLE
Unsupported python versions for particular linters - not flagged by tests?

### ISSUE THREAD

**Issue by amyheather** [lintquarto/lintquarto#42]
Although lintquarto supports Python 3.7 to 3.13, certain linters may not.

For example, `pytype` failed on Python 3.13 when I tried on vidigi documentation.

How come I didn't get flagged by this from the tests via GitHub actions?

You can also see this issue in the lintquarto documentation.

- [x] Add test/s that flag up this issue. Test did not flag issue as the test running `lintquarto` was only running `flake8`. Amend to run on all.
- [x] Amend the python version used to build the documentation
- [ ] Add logic that, instead of trying to run it all on your files and returning this error for every file, instead just says when you call it, `pytype` does not support Python 3.13, supported versions are ...
- [x] Don't run `pytype` tests on Python 3.13

```
=============================================================

Running pytype...

=============================================================

ninja: Entering directory `.pytype'

[1/1] check typecheck_example

FAILED: /home/runner/work/lintquarto/lintquarto/docs/pages/tools/.pytype/pyi/typecheck_example.pyi 

/opt/hostedtoolcache/Python/3.13.5/x64/bin/python -m pytype.main --imports_info /home/runner/work/lintquarto/lintquarto/docs/pages/tools/.pytype/imports/typecheck_example.imports --module-name typecheck_example --platform linux -V 3.13 -o /home/runner/work/lintquarto/lintquarto/docs/pages/tools/.pytype/pyi/typecheck_example.pyi --analyze-annotated --nofail --quick /home/runner/work/lintquarto/lintquarto/docs/pages/tools/examples/typecheck_example.qmd

Python versions > 3.12 are not yet supported.

ninja: build stopped: subcommand failed.

Computing dependencies

Analyzing 1 sources with 0 local dependencies

Leaving directory '.pytype'

```
